pyaqueduct/task.py

```diff
@@ -0,0 +1,89 @@
+"""Task module."""
+
+from uuid import UUID
+from datetime import datetime
+from typing import List
+
+from pydantic import BaseModel, PrivateAttr
+from pyaqueduct.client import AqueductClient
+from pyaqueduct.client.extension_types import ExtensionCancelResultData
+from pyaqueduct.client.task_types import ParameterData
+from pyaqueduct.client.experiment_types import ExperimentData
+
+
+class Task(BaseModel):
+    """Task representation. Contains all data and metadata about Aqueduct Task"""
+
+    _client: AqueductClient = PrivateAttr()
+    "Client object reference."
+
+    uuid: UUID
```

Comment:
Any reason for `uuid` against `task_id`?

Reply:
This was refactored in Aqueduct Core, hence this is valid now.